extensions/banner/deployment/src/main/java/io/quarkus/banner/deployment/BannerProcessor.java

```diff
@@ -21,7 +21,7 @@
 
     private static final Logger logger = Logger.getLogger(BannerProcessor.class);
 
-    @BuildStep(loadsApplicationClasses = true)
+    @BuildStep(loadsApplicationClasses = true, onlyIf = IsBanner.class)
```

Comment:
Not really a big fan of this pattern of injecting the config in the supplier. Well and it won't work if we want to make it runtime :).

Reply:
I on the other hand love it😎. To make it makes things very clear. Plus we use it all over the place.

Indeed if we decide to make the property a runtime one it will have to change